ravenframework/Optimizers/RavenSampled.py

```diff
@@ -560,6 +560,8 @@ def _resolveNewOptPoint(self, traj, rlz, optVal, info):
       # nothing else to do but wait for the grad points to be collected
     elif acceptable == 'rejected':
       self._rejectOptPoint(traj, info, old)
+      if self.printTag == 'GradientDescent':
+        self.__stepCounter[traj] -= 1
```

Comment:
this concerns me; are we not counting rejected steps as iterations now? This seems like it will create indexing problems and logistical mismatches. Help me understand why I'm wrong about that.